Cython/Compiler/ExprNodes.py

```diff
@@ -13749,8 +13785,15 @@ def __init__(self, pos, expr, string=None):
         if string is None:
             # import doesn't work at top of file?
             from .AutoDocTransforms import AnnotationWriter
-            string = StringEncoding.EncodedString(
-                AnnotationWriter(description="annotation").write(expr))
+            strval = AnnotationWriter(description="annotation").write(expr)
+            # For annotations which are already string literals, remove `u` prefix,
+            # since we're re-casting as a string and the result will be mangled
+            # ("u'foo'" instead of "'foo'")
+            # While still strictly compatible, this is not the same observed behavior
+            # as CPython in this instance.
+            if strval[0] == "u":
```

Comment:
I'm pretty sure `uint32` (variable name, not a string) could be a valid annotation that probably shouldn't be transformed into `int32`.

My view is that we shouldn't try to special-case this here. This all goes away in the medium future when Python 2 support is dropped. It'd probably be possible to express the tests as an equality test with `eval` (e.g. `>>> eval(some_annotation) == "string of some sort")` should evaluate as `True`).

Reply:
Just merge `master` and remove this special case.

Reply:
Done - thanks!